### PR TITLE
feat: #53 Add filters on AllWays Page

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/service/WayService.ts [hunk 1]
```diff
@@ -73,9 +73,32 @@ export class WayService {
...
+      : [];
+    const isAbandonedConstraints = filter?.isAbandoned
+      ? [
+        where(WAY_IS_COMPLETED_FIELD, "==", false),
+        where(WAY_LAST_UPDATE_FIELD, "<", Timestamp.fromMillis(abandonedDate)),
+      ]
+      : [];
+
+    const constraints: QueryFieldFilterConstraint[] = [
+      ...isCompletedConstraints,
```
Comment (tritonJS826) [inline mw-webapp/src/service/WayService.ts:96]: we are using "is" prefix only for boolean values

File: mw-webapp/src/service/WayService.ts [hunk 2]
```diff
@@ -104,7 +145,10 @@ export class WayService {
 
     const startAfterConstraints = snapshot ? [startAfter(snapshot)] : [];
 
-    const constraints: (QueryOrderByConstraint | QueryLimitConstraint | QueryStartAtConstraint)[] = [
+    const constraints: (QueryFieldFilterConstraint | QueryOrderByConstraint | QueryLimitConstraint | QueryStartAtConstraint)[] = [
```
Comment (tritonJS826) [inline mw-webapp/src/service/WayService.ts:148]: should we create a separate time for constraints and reuse it?

File: mw-webapp/src/service/WayService.ts [hunk 3]
```diff
@@ -88,9 +111,27 @@ export class WayService {
   /**
    * Get WaysDTO
    */
-  public static async getWaysDTO(lastWayUuid?: string): Promise<WayDTO[]> {
+  public static async getWaysDTO(lastWayUuid?: string, filter?: GetWaysFilter): Promise<WayDTO[]> {
```
Comment (tritonJS826) [inline mw-webapp/src/service/WayService.ts:114]: its super hard to work with such interface, use object instaedd of two optional args

File: mw-webapp/src/service/WayService.ts [hunk 4]
```diff
@@ -73,9 +73,32 @@ export class WayService {
   /**
    * Get WaysDTO amount
    */
-  public static async getWaysDTOAmount(): Promise<number> {
+  public static async getWaysDTOAmount(filter?: GetWaysFilter): Promise<number> {
     const waysRef = collection(db, PATH_TO_WAYS_COLLECTION);
-    const snapshot = await getCountFromServer(waysRef);
+
+    const isCompletedConstraints = filter?.isCompleted ? [where(WAY_IS_COMPLETED_FIELD, "==", true)] : [];
```
Comment (tritonJS826) [inline mw-webapp/src/service/WayService.ts:79]: looks like thw whole block is copypaste

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx [hunk 5]
```diff
@@ -36,6 +55,26 @@ export const AllWaysPage = () => {
...
+
+    /**
+     * Check is stored data valid
+     */
+    storedDataValidator: (
+      currentSettings: AllWaysPageSettings,
+    ) => allWaysPageSettingsValidator(currentSettings),
+  });
+
+  const filter: GetWaysFilter | undefined = allWaysPageSettings.filterStatus !== FILTER_STATUS_ALL_VALUE
```
Comment (tritonJS826) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx:70]: this filter looks familiar for me, do we have the same one?

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx [hunk 6]
```diff
@@ -4,15 +4,34 @@ import {HorizontalContainer} from "src/component/horizontalContainer/HorizontalC
...
+
+/**
+ * Safe opened tab from localStorage
+ */
+const allWaysPageSettingsValidator = (currentSettings: AllWaysPageSettings) => {
+  if (!currentSettings.filterStatus) {
+    return false;
+  }
+
+  return true;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx:32]: use shorter syntax

**Review by tritonJS826 — APPROVED** [review]

**Review by Alex89198900 — APPROVED** [review]

**Pull request merged**